Shared/Extensions/Task+TaskPool.swift

```diff
@@ -18,9 +18,8 @@ class TaskPool {
         self.pool.append(task)
 
         Task { [weak self] in
-            guard let `self` = self else { return }
```

Comment:
Using the guard here creates a strong reference. The pool won't be deallocated while waiting for the task to finish. Hopefully this fixes the issue.